### PR TITLE
adding clean-up stage to script

### PR DESCRIPTION
**Pull request by kab163**
# Summary

- This PR is a  bugfix for gitlab CI
- It fixes issue that too many files are being generated during CI testing (and eventually takes up too much space)
- I also tried to add more 'date' commands
- It adds more time to CI as well (some tests were timing out)

Note:

Passing gitlab CI test suite is now ***REQUIRED*** to get your PR merged.
You can initiate the gitlab CI test suite by simply commenting on your PR 
after creation with something as simple as "LGTM".

### CODE REVIEW

**Comment by kab163** [comment]
LGTM

File: scripts/gitlab/build_and_test.sh
Comment (davidbeckingsale) [inline scripts/gitlab/build_and_test.sh:165]: I think you need to look up a directory for this one?

Comment (adrienbernede) [inline scripts/gitlab/build_and_test.sh:165]: The drawback of cleaning here is that the clean won't apply to any failing job.
You could put the clean commands in the "after_script" section of the CI job insteald.

Comment (kab163) [reply scripts/gitlab/build_and_test.sh]: (Resolving because new commit edits address this - uberenv_libs will be in the shared mem folder /dev/shm/ where it'll be deleted anyway)

**Comment by kab163** [comment]
LGTM

**Comment by kab163** [comment]
LGTM

File: scripts/gitlab/build_and_test.sh
Comment (davidbeckingsale) [inline scripts/gitlab/build_and_test.sh:174]: I think you need to be inside the build dir when you run make clean, it doesn't seem like it's doing anything right now.

Comment (kab163) [reply scripts/gitlab/build_and_test.sh:174]: gasp!

Comment (kab163) [reply scripts/gitlab/build_and_test.sh]: good catch, thanks

File: scripts/gitlab/build_and_test.sh
Comment (adrienbernede) [inline scripts/gitlab/build_and_test.sh:42]: Bash is hard, I know...
I found this in the logs:
```
scripts/gitlab/build_and_test.sh: line 42: [[-z: command not found
```
```suggestion
        if [[ -z ${job_unique_id} ]]; then
```

**Review by davidbeckingsale — DISMISSED** [review]

**Review by rhornung67 — APPROVED** [review]

File: scripts/gitlab/build_and_test.sh
Comment (adrienbernede) [inline scripts/gitlab/build_and_test.sh:48]:
```suggestion
          job_unique_id=manual_job_${date_in_sec}
```

**Review by adrienbernede — APPROVED** [review]

**Comment by adrienbernede** [comment]
@davidbeckingsale @rhornung67 I don't understand the behavior of approval dismissal.
Or, maybe it's a project specific setting?
Sorry for the inconvenience

**Comment by kab163** [comment]
> @davidbeckingsale @rhornung67 I don't understand the behavior of approval dismissal.
> Or, maybe it's a project specific setting?
> Sorry for the inconvenience

Lol I've done the same thing. It's a project setting somewhere - if there are any changes done after an approval, it makes it "stale". No worries, thanks for your updates!

File: scripts/gitlab/build_and_test.sh
Comment (davidbeckingsale) [inline scripts/gitlab/build_and_test.sh:44]: I don't think this is checking the correct location?

Comment (adrienbernede) [reply scripts/gitlab/build_and_test.sh:44]: Oops... of course! This explains why this was not working well even with the "sleep" thing.

Comment (adrienbernede) [reply scripts/gitlab/build_and_test.sh]: Fix, I think.

**Comment by rhornung67** [comment]
@kab163 is this ready to merge? @davidbeckingsale and @adrienbernede should approve first.

**Comment by kab163** [comment]
> @kab163 is this ready to merge? @davidbeckingsale and @adrienbernede should approve first.

I think it is, assuming those 2 approve, though the clean-up status messages may not be needed (https://github.com/LLNL/RAJA/blob/460526e712b8488cdd08d70c908702d7ffab1d7a/scripts/gitlab/build_and_test.sh#L172-L180)

File: scripts/gitlab/build_and_test.sh
Comment (adrienbernede) [inline scripts/gitlab/build_and_test.sh:180]: @kab163, I think this is not really useful in production, indeed. I understand the interest for debugging, but the information is less relevant in a daily use, and `du -sh` can be slow and introduce a delay in our pipelines.

**Review by adrienbernede — DISMISSED** [review]

File: scripts/gitlab/build_and_test.sh
Comment (rhornung67) [reply scripts/gitlab/build_and_test.sh:180]: Agreed. Running du in a directory with a lot of files/subdirectories can take a long time. Perhaps, leave these lines in the script for debugging if needed, but commented out so they don't run otherwise.

Comment (adrienbernede) [reply scripts/gitlab/build_and_test.sh:180]: I would prefer if those were removed. The commit removing those could be reverted at will, and the syntax is not very complex anyway.

Comment (kab163) [reply scripts/gitlab/build_and_test.sh:180]: Sounds good. Thanks for the feedback!

**Review by adrienbernede — APPROVED** [review]

**Review by rhornung67 — APPROVED** [review]
Looks good to me. Thanks @kab163

**Comment by kab163** [comment]
Great! Thanks guys. When ready, merge it!

**Comment by rhornung67** [comment]
@davidbeckingsale good to merge?

**Review by davidbeckingsale — APPROVED** [review]

**Comment by davidbeckingsale** [comment]
> @davidbeckingsale good to merge?

Yup, looks good.

**Pull request merged**
